docs/security-o2k.md

```diff
@@ -0,0 +1,119 @@
+# OpenAPI2Kong support for security directives
+
+This document documents the use of `security` directives within the `deck` subcommand `openapi2kong`. There are 2 sections;
+
+1. [OpenAPI 3 security directives explained](#openapi-3-security-directives-explained)
+2. [Conversion logic to Kong plugins](#conversion-logic-to-kong-plugins)
+
+---
+
+# Openapi 3 security directives explained
+
+Within OpenAPI the security directives can be specified on the document root (the `OpenAPI` object). It can also be specified on the `Operation` object, in which case it will override the document level one.
+Specifically; it cannot be specified on the `path` object level, the level in between document and operation.
+
+## Format
+
+The `security` property is an array of `Security Requirement` objects.
+
+## Behaviour
+
+- The requirements listed are a logical `OR`; passing any one requirement listed is enough to be allowed access.
+- The default value is `[]`; an empty array
+- An empty array allows access without authentication
+- An empty array (on the `Operation` level) can override a non-empty array (document level)
+
+## Security Requirement
+
+A `security requirement` object lists (as properties) `security scheme` objects that MUST all be satisfied to allow access.
+
+Example of a `security` directive listing 2 `security requirements`, each having 2 `security scheme` names :
+```yaml
+security:
+  - myOpenId: [ "scope3" ]
+    myBasicAuth: []
+  - myBasicAuth: []
+    myKeyAuth: []
+```
+To authenticate a request must either pass:
+- `myOpenId` (with 'scope3') **AND** `myBasicAuth`, or
+- `myBasicAuth` **AND** `myKeyAuth`
+
+**NOTE**: the values (the arrays), contain scope names to use for `oauth2` and `openIdConnect` security schemes.
+(for all other types the array must be empty).
+
+## Security Scheme
+
+The `security scheme` object specifies a single security/authication mechanism to validate. The schemes are defined as properties on the `components.securitySchemes` object.
```

Comment:
```suggestion
The `security scheme` object specifies a single security/authentication mechanism to validate. The schemes are defined as properties on the `components.securitySchemes` object.
```

Generally, i'm not sure we need to rebuild the OAS documentation in our repo docs.  The document [here](https://swagger.io/docs/specification/authentication/) provides good explanation of OAS security specification design.